tests/test_BlkDiagMatrix.py

```diff
@@ -0,0 +1,910 @@
+                      [ 0.37452234,  0.1789093 ]]),
+            np.array([[-0.06749134,  0.37452234],
+                      [ 0.37452234,  0.1789093 ]]),
+            np.array([[-0.13551364]]),
+            np.array([[-0.13551364]]),
+            np.array([[-0.18975111]]),
+            np.array([[-0.18975111]]),
+            np.array([[-0.22661312]]),
+            np.array([[-0.22661312]])
+        ])
+
+        self.blk_b = BlkDiagMatrix.from_blk_diag([
+            np.array([[-0.27457111, -0.33770709, -0.09067737,  0.52007584],
+                      [-0.33770709, -0.24677034,  0.20639731, -0.2078888 ],
+                      [-0.09067737,  0.20639731, -0.36507922, -0.37742705],
+                      [ 0.52007584, -0.2078888 , -0.37742705,  0.30641696]]),
+            np.array([[-0.05219921, -0.23403317, -0.46057571],
+                      [-0.23403317, -0.58669018,  0.02892691],
+                      [-0.46057571,  0.02892691, -0.01291359]]),
+            np.array([[-0.05219921, -0.23403317, -0.46057571],
+                      [-0.23403317, -0.58669018,  0.02892691],
+                      [-0.46057571,  0.02892691, -0.01291359]]),
+            np.array([[ 0.04180323,  0.03254542, -0.56899236],
+                      [ 0.03254542, -0.5865448 , -0.33459247],
+                      [-0.56899236, -0.33459247,  0.23441378]]),
+            np.array([[ 0.04180323,  0.03254542, -0.56899236],
+                      [ 0.03254542, -0.5865448 , -0.33459247],
+                      [-0.56899236, -0.33459247,  0.23441378]]),
+            np.array([[ 0.02685116,  0.25494821],
+                      [ 0.25494821, -0.33546046]]),
+            np.array([[ 0.02685116,  0.25494821],
+                      [ 0.25494821, -0.33546046]]),
+            np.array([[-0.04041477,  0.36934457],
+                      [ 0.36934457,  0.0278976 ]]),
+            np.array([[-0.04041477,  0.36934457],
+                      [ 0.36934457,  0.0278976 ]]),
+            np.array([[-0.11861509,  0.37524325],
+                      [ 0.37524325,  0.38311618]]),
+            np.array([[-0.11861509,  0.37524325],
+                      [ 0.37524325,  0.38311618]]),
+            np.array([[-0.18418421]]),
+            np.array([[-0.18418421]]),
+            np.array([[-0.22672867]]),
+            np.array([[-0.22672867]]),
+            np.array([[-0.24393745]]),
+            np.array([[-0.24393745]])
+        ])
+
+        self.blk_partition = self.blk_a.partition
+
+    def tearDown(self):
+        pass
+
+    def allallfunc(self, A, B, func=np.allclose):
+        """ checks assertTrue(func()) as it iterates through A, B """
+        for (a,b) in zip(A, B):
+            self.assertTrue(func(a,b))
+
+    def testBlkDiagMatrixCompat(self):
+        self.assertTrue(self.blk_a.check_compatible(self.blk_b))
+
+        # Create a differently shaped matrix
+        x = BlkDiagMatrix.from_blk_diag(self.blk_a[1:-1])
+        # code should raise
+        raised = False
+        try:
```

Comment:
Why not use `pytest.raises`?

Reply:
No reason. New to me. Changed. Nice syntax, thanks.